File: spring-cloud-contract/java-spring-cloud-contract/app-callee/src/contractTest/java/com/example/javaspringcloudcontract/ProducerBase.java
package com.example.javaspringcloudcontract;

import com.example.javaspringcloudcontract.presentation.CalleController;
import io.restassured.module.mockmvc.RestAssuredMockMvc;
import org.junit.jupiter.api.BeforeEach;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.verifier.messaging.boot.AutoConfigureMessageVerifier;

@AutoConfigureMessageVerifier
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
public class ProducerBase {
    @Autowired
    CalleController calleController;

    @BeforeEach
    void setup() {
        RestAssuredMockMvc.standaloneSetup(calleController);
    }
}

File: spring-cloud-contract/app-callee/src/test/kotlin/com/example/springcloudcontract/ProducerBase.kt
package com.example.springcloudcontract

import com.example.springcloudcontract.presentation.TestApiController
import io.restassured.module.mockmvc.RestAssuredMockMvc
import org.junit.jupiter.api.BeforeEach
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.cloud.contract.verifier.messaging.boot.AutoConfigureMessageVerifier
import org.springframework.test.annotation.DirtiesContext

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
@DirtiesContext
@AutoConfigureMessageVerifier
open class ProducerBase {
    @Autowired
    lateinit var testApiController: TestApiController

    @BeforeEach
    fun setup() {
        RestAssuredMockMvc.standaloneSetup(testApiController)
    }
}